### PR TITLE
feat(cli): install pre-release binaries by computing binary URL from NPM URL

### LINKED ISSUE #8482: `cli` should be able to download correct prerelease binary version based on URL
### Current behavior:

Need to specify `CYPRESS_INSTALL_BINARY` when installing from prerelease .tgz.

### Desired behavior:

`CYPRESS_INSTALL_BINARY` can be autodiscovered if a prerelease .tgz is used.

### PR DESCRIPTION
**Pull request by flotwig**
- Closes #8482 

### User facing changelog

Installing Cypress pre-releases now no longer requires setting the `CYPRESS_BINARY_INSTALL` environment variable.

### How has the user experience changed?

Before, to install a pre-release:

```
export CYPRESS_INSTALL_BINARY=https://cdn.cypress.io/beta/binary/5.1.1/linux-x64/circle-issue-8482-78e05ab1227c842b2b9623bb583232c85ecabb82/cypress.zip
npm install https://cdn.cypress.io/beta/npm/5.1.1/circle-issue-8482-78e05ab1227c842b2b9623bb583232c85ecabb82/cypress.tgz
```

Now, to install a prerelease, no env var is needed:

```
npm install https://cdn.cypress.io/beta/npm/5.1.1/circle-issue-8482-78e05ab1227c842b2b9623bb583232c85ecabb82/cypress.tgz
```

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue been tagged with a release in ZenHub? 
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by chrisbreiding** [comment]
I don't know much about using pre-releases, so I'd be interested to see the 'How has the user experience changed' section elucidated. Also, is this documented somewhere?

**Comment by flotwig** [comment]
@chrisbreiding Elucidation complete! The docs are here for the existing system: https://docs.cypress.io/guides/getting-started/installing-cypress.html#Install-pre-release-version

I'll open a docs PR to update that once the implementation here is finalized.

File: cli/lib/tasks/install.js [hunk 1]
```diff
@@ -16,6 +17,89 @@ const unzip = require('./unzip')
...
 const { throwFormErrorText, errors } = require('../errors')
 
+const getNpmArgv = () => {
+  const json = process.env.npm_config_argv
+
+  if (!json) {
+    return
+  }
+
+  return JSON.parse(json).original || []
```
Comment (chrisbreiding) [inline cli/lib/tasks/install.js:27]: Any chance `json` could be invalid and cause parsing to throw?

Comment (flotwig) [reply cli/lib/tasks/install.js]: Don't think so, but I added a try/catch to be safe.

Comment (chrisbreiding) [reply cli/lib/tasks/install.js]: That was fast!

**Review by chrisbreiding — APPROVED** [review]
Looks good to me! I tried it out and it worked like a charm.

**Pull request merged**